### DISCUSSION TITLE
Welcome to pymgrid Discussions!

### DISCUSSION THREAD

**Discussion by GonzagueHenri** [Total-RD/pymgrid#63] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by TEKaal** [comment]
Hi! Currently, I am writing my master thesis and I am using this library! So many thanks. :) 

I have a question about the FixedLengthStochasticTrajectory(nr_steps) for my RL env. I cannot seem to get it working: it does take over a random initial step, but then it seems like it does not take over the final step in the correct way. 

Example: 

I take a trajectory length of 672 (a week with 15 min steps). When the training starts it resets and an initial step of 15725 is set and final step seems to be 16397. Total length of the data in this situation 17520 and the number of steps before it returns reward and starts a new episode is just until that number so little over 1000. 

Then it again selects a new initial step and end step, but then again it just runs until the end. 

Do you maybe have a suggestion of what might be going wrong?

Kind regards
